lib/saml_idp/xml_security.rb

```diff
@@ -41,45 +41,23 @@ def initialize(msg = nil, error_code = nil)
 
       C14N = 'http://www.w3.org/2001/10/xml-exc-c14n#'
       DSIG = 'http://www.w3.org/2000/09/xmldsig#'
-      DS_NS = { "ds" => DSIG }
+      DS_NS = { 'ds' => DSIG }
 
       attr_accessor :document
 
       def initialize(response)
         @document = Nokogiri.XML(response)
       end
 
-      def validate(idp_cert_fingerprint, soft = true, options = {})
-        log 'Validate the fingerprint'
-        base64_cert = find_base64_cert(options)
-        cert_text = Base64.decode64(base64_cert)
-
-        cert, error = valid_cert(cert_text)
-        if error.present?
+      def validate(idp_certificate, options = {}, soft: true)
```

Comment:
In general, I think changing the type (fingerprint to certificate) of an argument is risky! Like if this was a public API we'd want to like bump a major version, etc etc

For this gem where we tightly control the usage, it's probably ine

Reply:
oh agreed -- nothing that is called external to the gem actually changes (the XMLSecurity class is not called directly from the IdP -- IdP changes are all just updating the tests/removing the analytics code). 

I thought about just ditching the `validate` method and updating the code to use `validate_with_sha256` to avoid changing the type of an argument, but like you said, we control the usage, so it felt okay to do it this way.